boot/boot_serial/src/boot_serial.c

```diff
@@ -439,6 +439,41 @@ bs_upload(char *buf, int len)
     flash_area_close(fap);
 }
 
+#ifdef MCUBOOT_BOOT_MGMT_ECHO
+static bool 
+decode_echo(cbor_state_t *state, cbor_string_type_t *result)
+{
+	bool int_res;
+	size_t bsstrdecoded;
+
+	bool tmp_result = (((map_start_decode(state) && (int_res = (multi_decode(2, 2, &bsstrdecoded, (void *)tstrx_decode, state, result, sizeof(cbor_string_type_t))), ((map_end_decode(state)) && int_res)))));
```

Comment:
This is not readable.

Reply:
100% right :-). I was already wondering how this passed code review in the decode_Upload method.

Reply:
This looks like it was auto-generated by `cddl-gen`.